Bottom: 78aec4bb203000d649ade8edf068f3e82c472491
Top:    847b5b69562c87489d8f8787c4d792ad923bf0bc
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-06 16:27:59 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index b1283e6..75d197b 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -235,6 +235,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
